add kubernetes section (#456)
digital-asset/docs.daml.com: docs/2.8.0/docs/kubernetes/02-provision-cloud-resources/01-provision-cloud-resources.rst

.. Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
.. SPDX-License-Identifier: Apache-2.0

Provisioning Cloud Resources with Terraform
###########################################

.. note::

>    All the resources created will be private, nothing is being exposed over the public internet.

Reviewer: ```suggestion
   All the resources created are private, nothing is being exposed over the public internet.
```

* `Helm <https://helm.sh/docs/intro/install/>`_ [\ ``3.9+``\ ]
* `Helmfile (optional) <https://helmfile.readthedocs.io/>`_ [\ ``0.150+``\ ]

Steps
*****

.. note::
   All the below steps assume that your working directory is ``src/terraform``.

Create SSH Key for Accessing Bastion
====================================

You need to provide an existing SSH key pair to access the proxy (aka bastion) that will be created, you can create a new one using this command:

.. code-block:: sh

   ssh-keygen -m PEM -t rsa -b 4096 -f /path/to/ssh/key -C 'bastion@zero.k8s'

Set Up Your Azure Account
=========================

Log in to Azure and select the target subscription (if you have multiple ones):

.. code-block:: bash

   az login
   az account list
   az account set --subscription <subscription_id>

Configure Terraform
===================

To configure Terraform for your Azure subscription, follow the below steps:

* You need to provide an existing Azure resource group, you can create a new one using this command:

  .. code-block:: bash

     az group create --name <resource_group_name> --location <location>

* Create your own backend configuration

> * Copy and customize variables the file ``sample.tfvars``, use the resource group you just created, you can use the file name ``terraform.tfvars`` to avoid passing argument ``--var-file=/path/to/file.tfvars`` each run.

Author: I copied the sample.tfvars file and put it in the docs repo.
